code/modules/antagonists/xeno/xeno.dm

```diff
@@ -31,6 +31,56 @@
 /datum/antagonist/xeno/get_team()
 	return xeno_team
 
+/mob/living/carbon/alien/humanoid/royal/queen/Login()
+	..()
+	SSshuttle.registerHostileEnvironment(src)
+
+/mob/living/carbon/alien/humanoid/royal/queen/Logout()
+	..()
+	addtimer(CALLBACK(src, .proc/logoutcheck), 40 SECONDS)//when logging out, we don't want to spam announcements with leaving and reconnecting.
+	SSshuttle.clearHostileEnvironment(src)
+	neutralized = TRUE
```

Comment:
This will neutralise them if they logout, but not undo it if they log back in. Also I am pretty sure this should be in the xeno `Initialize` and `Destroy` procs.

Reply:
I tested it and it will create a hostile environment when they're logged in, and when they log out it clears it just fine. I'm not sure what you mean when you say ``Initialize`` and ``Destroy`` however.